### ISSUE TITLE
tree-sitter-typescript throws "Invalid language object" because it provides two grammars

### ISSUE THREAD

**Issue by DIbraDev** [dodona-edu/dolos#922]
Typescript parser is not working and error is not clear

```js
[error] Invalid language object
```
then I tried to compare files using Javascript parser, it worked for some files, but for others I got the below error:

```js
[error] Maximum call stack size exceeded
```

![image](https://user-images.githubusercontent.com/37841054/199927508-9a752fac-ddae-44eb-b37c-7e3a68c2a0a2.png)

can anyone help debugging this issue, please ?

node version: v18.12.0
npm version: 8.19.2
npm packages
├── @dodona/dolos@2.0.0
└── tree-sitter-typescript@0.20.1

**Comment by rien** [comment]
I haven't encountered the `Invalid language object` before, I'll try to investigate this issue later today.

How large are the files you are analyzing?

**Comment by rien** [comment]
The typescript error is cause by [tree-sitter-typescript](https://www.npmjs.com/package/tree-sitter-typescript) providing two grammars and requiring callers to explicitly specify which one to include.

This needs to be fixed in Dolos itself by adding a special case for typescript and tsx.

The error `Maximum call stack size exceeded` is probably caused by the two files being exceptionally big. Are you able to provide these two files so I can debug on those?

**Comment by DIbraDev** [comment]
> The typescript error is cause by [tree-sitter-typescript](https://www.npmjs.com/package/tree-sitter-typescript) providing two grammars and requiring callers to explicitly specify which one to include.
> 
> This needs to be fixed in Dolos itself by adding a special case for typescript and tsx.
> 
> The error `Maximum call stack size exceeded` is probably caused by the two files being exceptionally big. Are you able to provide these two files so I can debug on those?


Sure, you can test with those files [TS files.zip](https://github.com/dodona-edu/dolos/files/9937890/TS.files.zip), they are similar so we should get similarity 1.

Actually, my code packs all ts files into one to calculate global similarity.

Thanks in advance :)

**Comment by rien** [comment]
Thanks for the files. I was able to pin down why the maximum call stack error was thrown and fixed it with #924. Now Node itself is going out of memory though, so for this example here you might want to tweak the winnowing parameters:

With `dolos -l javascript -k 100 -w 100 pack1.ts pack2.ts` the analysis should already work. Setting `k` (kgram size) and `w` (window size) makes the analysis coarser (small differences are more likely to escape), but with such large files I assume you don't want to find these small differences anyway?

If you want, hit me up at rien.maertens@ugent.be to explain your use case a bit further. I would be happy to discuss how Dolos can be of help in your situation.
